aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttSubscriptionStatusCallback.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *    http://aws.amazon.com/apache2.0
+ *
+ * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazonaws.mobileconnectors.iot;
+
+/**
+ * Enables an application to be notified of status of call to subscribe to a topic.
+ */
+public interface AWSIotMqttSubscriptionStatusCallback {
+    /**
+     * This method is invoked when a susbcription has completed successfully.
+     */
+    void onSuccess();
```

Comment:
Are you sure there is no information required as part of the notification of success?

Reply:
There doesn't seem to be in this case. There binary information of whether subscription was successful. As far as the state of client is concerned, the client has to be connected for subscription to go through and we have a different callback tracking any state changes thereafter.